src/source-control/GitSourceControl.ts

```diff
@@ -1,47 +1,449 @@
+
+            // Stop if this is the common ancestor commit
+            if (nodegitCommit.id().equal(commonAncestorCommitOid)) {
+              stopped = true;
+            }
+          });
+
+          history.on("end", () => resolve());
+          history.start();
+        });
+      }),
+    );
+
+    // Build relationships
+    commitChildrenMap.forEach((childrenHashes, parentHash) => {
+      this.commitHashMap = produce(this.commitHashMap, (draftCommitHashMap) => {
+        const parentCommit = draftCommitHashMap.get(parentHash);
+        if (!parentCommit) {
+          return;
+        }
+        childrenHashes.forEach((childHash) => {
+          const childCommit: Commit = draftCommitHashMap.get(childHash)!;
+          childCommit.parentCommits = Array.from(
+            new Set([...childCommit.parentCommits, parentHash]),
+          );
+          parentCommit.childCommits = Array.from(
+            new Set([...parentCommit.childCommits, childHash]),
+          );
+        });
+      });
+    });
+
+    this.commitHashMap = produce(this.commitHashMap, (draftCommitHashMap) => {
+      refs.forEach((ref) => {
+        const branchName = ref.name();
+        const branchSha = ref.target().tostrS();
+        draftCommitHashMap.get(branchSha)?.refNames.push(branchName);
+      });
+    });
+
+    // Inject branch names
+
+    const hasUncommittedChanges = repoStatus.length > 0;
+    const earliestInterestingCommit = this.commitHashMap.get(
+      commonAncestorCommitOid.tostrS(),
+    )!;
+
+    const ourRepository: Repository = {
+      path: this.repoPath,
+      hasUncommittedChanges,
+      headHash,
+      earliestInterestingCommit,
+      commits: this.commitHashMap,
+    };
+
+    const getRemoteRepoInfo = async (
+      ourRepository: Repository,
+    ): Promise<Repository> => {
+      const remoteUpdatedCommitHashMap = await produce(
+        ourRepository.commits,
+        async (draftCommitHashMap) => {
+          // Get all commits with local branches
+          const allCommitsWithBranches = Array.from(
+            draftCommitHashMap.values(),
+          ).filter((commit) => commit.refNames.filter(refIsLocal).length !== 0);
+
+          // Look up PR information for them and inject their information
+          await Promise.all(
+            allCommitsWithBranches.map(async (commit) => {
+              // TODO: Implement this.getPullRequestInfo
+              commit.pullRequestInfo = await this.getPullRequestInfo(
+                this.repoPath,
+                commit.hash,
+                // TODO: Handle commits with multiple PRs
+                commit.refNames.filter(refIsLocal)[0],
+              );
+            }),
+          );
+        },
+      );
+      const updatedRepository = {
+        ...ourRepository,
+        commits: remoteUpdatedCommitHashMap,
+      };
+      return updatedRepository;
+    };
+
+    return {
+      repo: ourRepository,
+      remoteRepoInfoPromise: getRemoteRepoInfo(ourRepository),
```

Comment:
We no longer need this promise since it’s now Stacker’s job to get PR info (cc @manyaagarwal)